Content.Shared/Xenoarchaeology/Artifact/XAT/XATDeathSystem.cs

```diff
@@ -0,0 +1,46 @@
+using Content.Shared.Mobs;
+using Content.Shared.Xenoarchaeology.Artifact.Components;
+using Content.Shared.Xenoarchaeology.Artifact.XAT.Components;
+
+namespace Content.Shared.Xenoarchaeology.Artifact.XAT;
+
+public sealed class XATDeathSystem : BaseXATSystem<XATDeathComponent>
+{
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+
+    private EntityQuery<XenoArtifactComponent> _xenoArtifactQuery;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        _xenoArtifactQuery = GetEntityQuery<XenoArtifactComponent>();
+
+        SubscribeLocalEvent<MobStateChangedEvent>(OnMobStateChanged);
+    }
+
+    private void OnMobStateChanged(MobStateChangedEvent args)
+    {
+        if (args.NewMobState != MobState.Dead)
+            return;
+
+        var targetCoords = Transform(args.Target).Coordinates;
+
+        var query = EntityQueryEnumerator<XATDeathComponent, XenoArtifactNodeComponent>();
+        while (query.MoveNext(out var uid, out var comp, out var node))
```

Comment:
There's quite a few "if this ever happens, uh, try and trigger this even if it's very far away" effects. This could be messy in some extreme cases.

Reply:
Elaborate, pls... is this actionable or just the idea of something bad happening in XAT systems?

Reply:
Any more details there? >_> or ideas?